### PR TITLE
Add option to display config enums as radio buttons

### PR DESCRIPTION
**Pull request by marnen**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

This pull request adds support for a `radio` option in the config schema, for `enum` config values. If `radio` is `true`, then the enum will be displayed as a group of radio buttons. 

*Edit by @rsese to add screenshot from https://github.com/atom/settings-view/pull/1089#issuecomment-435422204*

image

### Alternate Designs

I chose to add an option here because it seemed like the easiest way to not break existing config files, and to maintain backwards compatibility for new config files. If a config file using the `radio: true` feature is processed by an older version of Atom, the extra key will simply be ignored and the option will be displayed as a `<select>` element.

### Benefits

For small numbers of options, radio buttons often provide a better user experience than a drop-down menu would. This is especially true if the option descriptions are on the long side.

*Edit by @rsese to add extra context from https://github.com/atom/settings-view/pull/1089#issuecomment-435248703*

If there are fewer than approximately 5 options, radio buttons show them all at a glance, without needing to click. Likewise, it takes only 1 click to select a radio button, as opposed to 2 clicks for a drop-down option. There are other reasons too, but those are the most prominent.

### Possible Drawbacks

I designed this to be as free of negative impact as I could manage.

### CODE REVIEW

File: lib/settings-panel.js [hunk 1]
```diff
@@ -322,6 +326,13 @@ export default class SettingsPanel extends CollapsibleSectionPanel {
       } else {
         return floatValue
       }
+    } else if (type === 'integer') {
```
Comment (marnen) [inline lib/settings-panel.js]: I might be able to unify this code with current float-parsing logic.

**Comment by marnen** [comment]
Looks like the only failing test on AppVeyor is one that's been failing intermittently anyway, and so I believe it's unrelated to this pull request.

**Comment by rsese** [comment]
Thanks @marnen! 

Can you share a screenshot of what this look like with your pull request?

Also:

> For small numbers of options, radio buttons often provide a better user experience than a drop-down menu would. This is especially true if the option descriptions are on the long side.

Can you share an example of this that shows why radio buttons would be better for a particular option?

**Comment by marnen** [comment]
@rsese Sure, I’ll share a screenshot later when I’m back at my computer. 

> Can you share an example of this that shows why radio buttons would be better for a particular option?

If there are fewer than 5 or so options, radio buttons show them all at a glance, without needing to click. Likewise, it takes only 1 click to select a radio button, as opposed to a drop-down option. There are other reasons too, but those are the most prominent.

**Comment by marnen** [comment]
See also https://blog.prototypr.io/7-rules-of-using-radio-buttons-vs-drop-down-menus-fddf50d312d1 (which I just found this minute). It reinforces what I was saying, but more clearly.

**Comment by marnen** [comment]
Here's the promised screenshot. 
image

**Comment by rsese** [comment]
Thanks!  I added the screenshot and the extra explanation of the benefits to the issue body and  someone from the team will take a look as soon as they can.

**Comment by lee-dohm** [comment]
Would you mind fixing the merge conflict so we can review this?

**Comment by marnen** [comment]
Yikes! I wasn’t aware that there was one; sorry about that. Apparently the mobile interface for GitHub doesn’t show that piece of info. I’ll be happy to.

**Comment by marnen** [comment]
@lee-dohm Should be resolved now.

**Comment by marnen** [comment]
I see some linter failures. I’ll fix those.

**Comment by marnen** [comment]
Really? The linter wants a space between the function name and the parentheses? That seems very odd to me...

**Comment by marnen** [comment]
This should be ready to merge now.

**Comment by marnen** [comment]
Just checking: is there anything further preventing this from being reviewed?

**Comment by rsese** [comment]
@marnen - sorry it's just a matter of workload, but when one of our engineers can review you'll see comments/questions/etc.

**Comment by marnen** [comment]
Sure, understood. I just wanted to make sure that there was nothing else procedural that I had to take care of.

**Comment by nathansobo** [comment]
I'm assuming you adding explicit parsing for the `integer` type because before we were allowing it to pass through as a string, which breaks logic associated with choosing the correct radio button? Just going to merge based on that assumption. Seems good to parse the integer regardless.

Thanks for the contribution and sorry for the delay.

**Comment by marnen** [comment]
@nathansobo After 5 months, I’m afraid I don’t remember exactly why I put the parsing in, but I do believe that it was as you say, because choosing the correct radio button wasn’t working. 

Thanks for merging this! I believe this will help make the settings panel UI a bit more pleasant.

File: lib/settings-panel.js (hunk 1 above)
Comment (people2963) [reply lib/settings-panel.js]: Any assistance u can offer to help me set up & download PKCS#12.pfx certificates

**Pull request merged**